source/linux/epoll_event_loop.c

```diff
@@ -407,11 +407,10 @@ static int s_subscribe_to_io_events(
         AWS_LOGF_ERROR(
             AWS_LS_IO_EVENT_LOOP, "id=%p: failed to subscribe to events on fd %d", (void *)event_loop, handle->data.fd);
         aws_mem_release(event_loop->alloc, epoll_event_data);
+        handle->additional_data = NULL;
```

Comment:
I would be EVER-SO-SLIGHTLY more comfortable if you move this above the mem release here, just because there's now an ever-so-slight chance that another thread or could see `handle->additional_data` point at freed memory.

Reply:
ok